common/stream-model.cpp

```diff
@@ -780,177 +774,213 @@ namespace rs2
-                    "Timestamp is measured in milliseconds, and is allowed to roll-over (reset to zero) in some situations" });
-                stream_details.push_back({ "Clock Domain",
-                    rsutils::string::from() << rs2_timestamp_domain_to_string(timestamp_domain),
-                    "Clock Domain describes the format of Timestamp field. It can be one of the following:\n"
-                    "1. System Time - When no hardware timestamp is available, system time of arrival will be used.\n"
-                    "                 System time benefits from being comparable between device, but suffers from not being able to approximate latency.\n"
-                    "2. Hardware Clock - Hardware timestamp is attached to the frame by the device, and is consistent accross device sensors.\n"
-                    "                    Hardware timestamp encodes precisely when frame was captured, but cannot be compared across devices\n"
-                    "3. Global Time - Global time is provided when the device can both offer hardware timestamp and implements Global Timestamp Protocol.\n"
-                    "                 Global timestamps encode exact time of capture and at the same time are comparable accross devices." });
-                stream_details.push_back({ "Frame Number",
-                    rsutils::string::from() << frame_number, "Frame Number is a rolling ID assigned to frames.\n"
-                    "Most devices do not guarantee consequitive frames to have conseuquitive frame numbers\n"
-                    "But it is true most of the time" });
-
-                if (profile.as<rs2::video_stream_profile>())
-                {
-                    stream_details.push_back({ "Hardware Size",
-                        rsutils::string::from() << original_size.x << " x " << original_size.y,
-                        "Hardware size is the original frame resolution we got from the sensor, before applying post processing filters." });
+    void stream_model::draw_text_background( const rect stream_rect, const float num_of_lines, const float max_text_width )
+    {
+        ImGui::GetWindowDrawList()->AddRectFilled(
+            { stream_rect.x - 5, stream_rect.y },
+            { stream_rect.x + max_text_width * 2,
+              stream_rect.y + ( num_of_lines * ( ImGui::GetTextLineHeight() + 3.5f ) ) },
+            ImColor( alpha( sensor_bg, 0.2f ) ) );
+    }
 
-                    stream_details.push_back({ "Display Size",
-                        rsutils::string::from() << size.x << " x " << size.y,
-                        "When Post-Processing is enabled, the actual display size of the frame may differ from original capture size" });
-                }
-                stream_details.push_back({ "Pixel Format",
-                    rsutils::string::from() << rs2_format_to_string(profile.format()), "" });
+    void stream_model::start_show_metadata( const long long timestamp,
```

Comment:
No need for a function just to call another function.
Better to combine to a single 
`draw_stream_metadata` function

Reply:
- The function `start_show_metadata` will be renamed to `draw_stream_metadata`.
- A code from the small function `draw_text_background` will be moved to `draw_stream_metadata` function.

Reply:
`draw_text_background` is OK to keep, up to you.
BTW I played with the viewer and it look good :)

Reply:
Do the logic in 1 function instead of calling one that calls another
If you are not sure call me and we'll talk :)